Benchmarks table

gabrielbosio opened the pull request.

gabrielbosio committed:
Vary benches according to security level

gabrielbosio committed:
Merge branch 'main' into vary_security_in_benches

gabrielbosio committed:
Separate benches

gabrielbosio committed:
Merge branch 'main' into vary_security_in_benches

gabrielbosio committed:
Arrange toml

gabrielbosio committed:
Ignore more compiled files

gabrielbosio committed:
Refactor big fibo benches

gabrielbosio committed:
Add the necessary to build benchmark table

gabrielbosio committed:
Add trace building benchmarks

gabrielbosio committed:
Add proof measurement

gabrielbosio committed:
Fix compilation

gabrielbosio committed:
Fix format

gabrielbosio committed:
Add table for M1

gabrielbosio committed:
Add table for Intel Xeon

MauroToscano commented on README.md:
```diff
@@ -139,3 +139,124 @@ if you don´t have the tools for fuzzing installed use
+
+First table has the results that are independent of the hardware used.
+
+| n   | Trace length | Trace time | Proof size 80 | Proof size 128 |
+|-----|--------------|------------|---------------|----------------|
+| 100 | 2^10         | 0.9 ms     | 270 KB        | 1.2 MB         |
+| 500 | 2^12         | 5.3 ms     | 335 KB        | 1.5 MB         |
+| 2k  | 2^14         | 24.7 ms    | 407 KB        | 1.8 MB         |
+| 5k  | 2^16         | 77.2 ms    | 488 KB        | 2.2 MB         |
+| 20k | 2^18         | 312 ms     | 576 KB        | 2.6 MB         |
+
```
It's not clear 80 and 128 are security bits, we should add that to the text

We should make one of conjecturable, and another one of provable. This is mixing both and is confusing

MauroToscano replied:
Since others projects are benchmarking with conjecturable, we should do that too, the other one is not necessary for this public data

gabrielbosio committed:
Compare conjecturable security levels

Intel measurements are not updated yet.

gabrielbosio committed:
Merge branch 'main' into vary_security_in_benches

gabrielbosio committed:
Add verification assertion

gabrielbosio committed:
Update benchmark table

Update Intel verifying time results.
Separate trace time into Apple and Intel.
Add info about bits of security.

gabrielbosio committed:
Add prover RAM usage benches

gabrielbosio committed:
Fix format
